schema.cc

```diff
@@ -638,6 +640,141 @@ std::ostream& operator<<(std::ostream& os, const schema& s) {
     return os;
 }
 
+std::ostream& map_as_cql_param(std::ostream& os, const std::map<sstring, sstring>& map, bool first = true) {
+    for (auto i: map) {
+        if (first) {
+            first = false;
+        } else {
+            os << ",";
+        }
+        os << "'" << i.first << "': '" << i.second << "'";
+    }
+
+    return os;
+}
+
+std::ostream& column_definition_as_cql_key(std::ostream& os, const column_definition & cd) {
+    os << cd.name_as_cql_string();
+    os << " " << cd.type->cql3_type_name();
+
+    if (cd.kind == column_kind::static_column) {
+        os << " STATIC";
+    }
+    return os;
+}
+
+static bool is_secondary_index(const utils::UUID& id, const schema& s) {
+    return  service::get_local_storage_service().db().local().find_column_family(id).get_index_manager().is_global_index(s);
```

Comment:
You left out local indexes. Your patch is incomplete without recognizing those, and also recreating them with the proper CQL command. And of course, you need to add a test case for them, which will show you you missed them. I already pointed out this in my previous round of review - see my other comment below which remains.